File: test-suite-docs-python/src/test/python/micronaut/mcp/docs/tools/ToolsFactoryTest.py
import json
from typing import Annotated

from jakarta.inject import Inject
from micronaut.context.annotation import Property
from micronaut.http import HttpRequest
from micronaut.http.client import HttpClient
from micronaut.http.client.annotation import Client
from micronaut.mcp.docs.JsonRpcMessages import EXPECTED_TOOLS_CALL, EXPECTED_TOOLS_LIST, TOOLS_CALL, TOOLS_LIST
from micronaut.test.extensions.junit5.annotation import MicronautTest
from org.junit.jupiter.api import Test


@Property(name="micronaut.mcp.server.transport", value="HTTP")
@Property(name="spec.name", value="ToolsFactoryTest")
@MicronautTest
class ToolsFactoryTest:
    client: Annotated[HttpClient, Inject, Client("/")]

    @Test
    def tools_list(self) -> None:
        result = str(self.client.toBlocking().retrieve(HttpRequest.POST("/mcp", TOOLS_LIST)))
        assert json.loads(result) == json.loads(EXPECTED_TOOLS_LIST)

    @Test
    def tools_call(self) -> None:
        result = str(self.client.toBlocking().retrieve(HttpRequest.POST("/mcp", TOOLS_CALL)))
        assert json.loads(result) == json.loads(EXPECTED_TOOLS_CALL)

File: test-suite-docs-java/src/test/java/io/micronaut/mcp/docs/tools/ToolsFactoryTest.java
package io.micronaut.mcp.docs.tools;

import io.micronaut.context.annotation.Property;
import io.micronaut.http.HttpRequest;
import io.micronaut.http.client.BlockingHttpClient;
import io.micronaut.http.client.HttpClient;
import io.micronaut.http.client.annotation.Client;
import io.micronaut.test.extensions.junit5.annotation.MicronautTest;
import org.json.JSONException;
import org.junit.jupiter.api.Test;
import org.skyscreamer.jsonassert.JSONAssert;

import static io.micronaut.mcp.docs.JsonRpcMessages.EXPECTED_TOOLS_CALL;
import static io.micronaut.mcp.docs.JsonRpcMessages.EXPECTED_TOOLS_LIST;
import static io.micronaut.mcp.docs.JsonRpcMessages.TOOLS_CALL;
import static io.micronaut.mcp.docs.JsonRpcMessages.TOOLS_LIST;
import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;

@Property(name = "micronaut.mcp.server.transport", value = "HTTP")
@Property(name = "spec.name", value = "ToolsFactoryTest")
@MicronautTest
class ToolsFactoryTest {

    @Test
    void toolsList(@Client("/") HttpClient httpClient) throws JSONException {
        BlockingHttpClient client = httpClient.toBlocking();
        String result = assertDoesNotThrow(() -> client.retrieve(HttpRequest.POST("/mcp", TOOLS_LIST)));
        JSONAssert.assertEquals(EXPECTED_TOOLS_LIST, result, true);
    }

    @Test
    void toolsCall(@Client("/") HttpClient httpClient) throws JSONException {
        BlockingHttpClient client = httpClient.toBlocking();
        String result = assertDoesNotThrow(() -> client.retrieve(HttpRequest.POST("/mcp", TOOLS_CALL)));
        JSONAssert.assertEquals(EXPECTED_TOOLS_CALL, result, true);
    }
}
